tdmpc2/tdmpc2.py

```diff
@@ -86,7 +102,10 @@ def act(self, obs, t0=False, eval_mode=False, task=None):
 			task = torch.tensor([task], device=self.device)
 		z = self.model.encode(obs, task)
 		if self.cfg.mpc:
-			a = self.plan(z, t0=t0, eval_mode=eval_mode, task=task)
+			kwargs = {}
+			if task is not None:
+				kwargs["task"] = task
```

Comment:
here too, plan won't like non-tensor args so if task is None I just don't pass it